src/strategy/__tests__/redis_cache_provider.test.ts

```diff
@@ -71,63 +71,72 @@ describe("getExtendedRedisCacheProvider#save", () => {
       callback(null, "OK")
     );
     const redisCacheProvider = getExtendedRedisCacheProvider(mockRedisClient);
-    const cacheSAMLResponse = await redisCacheProvider
-      .save(SAMLRequest, samlConfig)
-      .run();
+    const cacheSAMLResponse = await redisCacheProvider.save(
+      SAMLRequest,
+      samlConfig
+    )();
     expect(mockSet.mock.calls[0][0]).toBe(`SAML-EXT-${expectedRequestID}`);
     expect(mockSet.mock.calls[0][1]).toEqual(expect.any(String));
     expect(mockSet.mock.calls[0][2]).toBe("EX");
     expect(mockSet.mock.calls[0][3]).toBe(keyExpirationPeriodSeconds);
     expect(isRight(cacheSAMLResponse)).toBeTruthy();
-    expect(cacheSAMLResponse.value).toEqual({
-      RequestXML: SAMLRequest,
-      createdAt: expect.any(Date),
-      idpIssuer: samlConfig.idpIssuer
-    });
+    if (isRight(cacheSAMLResponse)) {
+      expect(cacheSAMLResponse.right).toEqual({
+        RequestXML: SAMLRequest,
+        createdAt: expect.any(Date),
+        idpIssuer: samlConfig.idpIssuer
+      });
+    }
   });
   it("should return an error if save on radis fail", async () => {
     const expectedRedisError = new Error("saveError");
     mockSet.mockImplementation((_, __, ___, ____, callback) =>
       callback(expectedRedisError)
     );
     const redisCacheProvider = getExtendedRedisCacheProvider(mockRedisClient);
-    const cacheSAMLResponse = await redisCacheProvider
-      .save(SAMLRequest, samlConfig)
-      .run();
+    const cacheSAMLResponse = await redisCacheProvider.save(
+      SAMLRequest,
+      samlConfig
+    )();
     expect(mockSet.mock.calls[0][0]).toBe(`SAML-EXT-${expectedRequestID}`);
     expect(mockSet.mock.calls[0][1]).toEqual(expect.any(String));
     expect(mockSet.mock.calls[0][2]).toBe("EX");
     expect(mockSet.mock.calls[0][3]).toBe(keyExpirationPeriodSeconds);
     expect(isRight(cacheSAMLResponse)).toBeFalsy();
-    expect(cacheSAMLResponse.value).toEqual(
-      new Error(
-        `SAML#ExtendedRedisCacheProvider: set() error ${expectedRedisError}`
-      )
-    );
+    if (isRight(cacheSAMLResponse)) {
+      expect(cacheSAMLResponse.right).toEqual(
```

Comment:
We expect that `cacheSAMLResponse` should be `left` (`expect(isRight(cacheSAMLResponse)).toBeFalsy();`) so we need to check the left path 
```typescript
if (isLeft(cacheSAMLResponse)) {
      expect(cacheSAMLResponse.left).toEqual(
```

Reply:
You are right, thank you!